### DISCUSSION TITLE
classes/class_itemlist

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#180] [User-contributed notes]
# classes/class_itemlist

Inherits: Control< CanvasItem< Node< Object A vertical list of selectable items with one or multiple columns. Description: This control provides a vertical list of selectable items that may be in a...

https://docs.godotengine.org/en/stable/classes/class_itemlist.html

**Comment by FoxyFox909** [comment]
As of 4.4.dev3: If you're going to use `ItemList.get_v_scroll_bar` in a `_ready` callback, then you should do it in a script attached to the `ItemList` Node in question AND await a process frame, or you won't be able to properly access the internal Node's properties.

Also, be careful when calling `ItemList.get_v_scroll_bar` a second time. It appears to have the side effect of instantiating and returning a *separate* Control Node.

Rather, just call this method once in the _ready function of the `ItemList` Node, do what you gotta do with the `VScrollBar` in the same script, and if you need to access it elsewhere, save a reference to it and access it through that variable instead of calling `get_v_scroll_bar` again

**Reply by FoxyFox909** [reply]
Example code:
```GDScript
extends ItemList

func _ready() -> void:
	await get_tree().process_frame
	var bar : VScrollBar = get_v_scroll_bar()
	bar.hide() ## Will succesfully hide the VScrollBar.
```

```GDScript
extends ItemList

func _ready() -> void:
	var bar : VScrollBar = get_v_scroll_bar()
	bar.hide() ## Will fail to hide the VScrollBar.
```

**Comment by stayinsidegames** [comment]
Although the description doesn't mention it, and ItemList does not inherit from Container or ScrollContainer, ItemList will automatically add a scrollbar and allow scrolling when the list is longer than the ItemList height, just like ScrollContainer.
